proc_blocks/fft/src/lib.rs

```diff
@@ -49,24 +49,62 @@ impl Fft {
 
     fn transform_inner(&mut self, input: Vec<i16>) -> [u8; 1960] {
         // Build the spectrogram computation engine
-        let mut spectrograph = SpecOptionsBuilder::new(49, 40)
-        .load_data_from_memory(input, self.sample_rate)
+        let mut spectrograph = SpecOptionsBuilder::new(49, 241)
+        .set_window_fn(sonogram::hann_function)
+        .load_data_from_memory(input, self.sample_rate as u32)
         //.unwrap()
         .build();
 
-        // // Compute the spectrogram giving the number of bins and the window
-        // // overlap.
-        spectrograph.compute(self.bins, self.window_overlap);
-
-        let result_f32 = spectrograph.create_in_memory(false);
-
-        let min_value = result_f32.iter().fold(f32::INFINITY, |a, &b| a.min(b));
-        let max_value =
-            result_f32.iter().fold(f32::NEG_INFINITY, |a, &b| a.max(b));
+        // Compute the spectrogram giving the number of bins in a window and the
+        // overlap between neighbour windows.
+        spectrograph.compute(480, 0.66666666666666667);
+
+        let spectrogram_f32 = spectrograph.create_in_memory(false);
+
+        let filter_count: usize = 40;
+        let power_spectrum_size = 241;
+        let window_size = 480;
+
+        // build up the mel filter matrix
+        let mut mel_filter_matrix =
+            DMatrix::<f64>::zeros(filter_count, power_spectrum_size);
+        for (row, col, coefficient) in mel::enumerate_mel_scaling_matrix(
+            self.sample_rate,
+            window_size,
+            power_spectrum_size,
+            filter_count,
+        ) {
+            mel_filter_matrix[(row, col)] = coefficient;
+        }
 
-        let res: Vec<u8> = result_f32
-            .into_iter()
-            .map(|freq| 255.0 * (freq - min_value) / (max_value - min_value))
+        let doubles = spectrogram_f32.into_iter().map(f64::from);
+        let power_spectrum_matrix_unflipped: DMatrix<f64> =
+            DMatrix::from_iterator(49, power_spectrum_size, doubles);
+        let power_spectrum_matrix_transposed =
+            power_spectrum_matrix_unflipped.transpose();
+        let mut power_spectrum_vec: Vec<_> =
+            power_spectrum_matrix_transposed.row_iter().collect();
+        &power_spectrum_vec.reverse();
+        let power_spectrum_matrix: DMatrix<f64> =
+            DMatrix::from_rows(&power_spectrum_vec);
+        let mel_spectrum_matrix = &mel_filter_matrix * &power_spectrum_matrix;
+
+        let min_value = mel_spectrum_matrix
+            .data
+            .as_vec()
+            .iter()
+            .fold(f64::INFINITY, |a, &b| a.min(b));
+        let max_value = mel_spectrum_matrix
+            .data
+            .as_vec()
+            .iter()
+            .fold(f64::NEG_INFINITY, |a, &b| a.max(b));
```

Comment:
It looks like `nalgebra::base::Matrix` has `min()` and `max()` methods.

```suggestion
        let min_value = mel_spectrum_matrix.min();
        let max_value = mel_spectrum_matrix.max();
```